loaders/authentication.js

```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch');
+const dotenv = require('dotenv').config();
+
+//base URL of cigarra authentication
+const URL = 'https://sigarra.up.pt/feup/pt/mob_val_geral.autentica?'
+
+//build URL with credentials
+let urlRequest = URL + 'pv_login=' + process.env.UP_USER + '&pv_password=' + process.env.UP_PWD;
+
+let cookie = null;
+
+//Called sigarra api to make authentication
+async function updateCookie(){
+   try {
+        const response = await fetch(urlRequest, {
+            method: 'POST'
+        });
+        if(response.ok){
+            return await response.json();
+        }
+        throw new Error('Request Failed: ' + response.status + "-" + response.statusText);
```

Comment:
This error is thrown and never caught. It will most likely crash the server. This would make sense since if authentication is not possible, this project will not be fully capable of working. However, crashing may be too drastic and we might just want to show some debug messages? I'm really not sure what would be the expected course of action here. @GambuzX thoughts?

Reply:
Isn't it being caught in line 23 tho?

Reply:
The code has since been changed to include a try catch around the block that throws an exception 🙂 

Still, this does not do anything special. If we are catching the exception straight away, it's like we aren't even throwing it.

My point was to start discussion on what would be the expected behaviour for this section of the codebase. Should the server crash in case of invalid credentials? Or should its functionality just be hampered?

I think that the expected behaviour (for me) would be that if the login was unsuccessful an error should be logged saying something like "configured credentials are invalid" and silently return null from the cookie function.  
That way the endpoints that need to use this could notice that the cookie was empty and just send a response with status 403 straight away.

Reply:
That makes sense. The login API may be down for some reason and it would be bad to crash TTS because of that.
That way we can notice if something ever goes wrong with the login API

Reply:
There is something that is not clear for me. If I remove the throw statement do I still need the try catch?